src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -532,10 +546,10 @@ public void RegisterConnectResult(SocketError error)
             }
         }
 
-        private bool TryBeginOperation<TOperation>(ref OperationQueue<TOperation> queue, TOperation operation, Interop.Sys.SocketEvents events, bool maintainOrder, out bool isStopped)
+        private bool TryBeginOperation<TOperation>(ref OperationQueue<TOperation> queue, object queueLock, TOperation operation, Interop.Sys.SocketEvents events, bool maintainOrder, out bool isStopped)
```

Comment:
This bothered me as well, especially since most of the callers actually already hold this lock.  I've restructured it to make it the caller's responsibility.